### PR TITLE
add time step masking (refs #477)

### PR DESCRIPTION
**Pull request by cwmeijer**
Add basic mean masking for timeseries.
@geek-yang already reviewed most of it.

### CODE REVIEW

**Comment by cwmeijer** [comment]
Linter issues reported by Github actions are in the dashboard and have already been fixed in main since this run.

File: dianna/utils/maskers.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    """Calculates a masking value of the given type for the data."""
+    if mask_type == 'mean':
+        return np.mean(data)
+    raise ValueError(f'Unknown mask_type selected: {mask_type}')
+
+
+def _determine_number_of_steps_masked(p_keep: float, series_length: int) -> int:
+    user_requested_steps = int(np.round(series_length * (1 - p_keep)))
+    if user_requested_steps == series_length:
+        warnings.warn('Warning: p_keep chosen too low. Continuing with masking 1 time step per mask.')
```
Comment (geek-yang) [inline dianna/utils/maskers.py:61]:
```suggestion
        warnings.warn('Warning: p_keep chosen too low. Continuing with leaving 1 time step unmasked per mask.')
```
Should be the opposite.

File: dianna/utils/maskers.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        return series_length - 1
+    if user_requested_steps == 0:
+        warnings.warn('Warning: p_keep chosen too high. Continuing with leaving 1 time step unmasked per mask.')
```
Comment (geek-yang) [inline dianna/utils/maskers.py:64]:
```suggestion
        warnings.warn('Warning: p_keep chosen too high. Continuing with masking 1 time step per mask.')
```
Same here.

**Review by geek-yang — APPROVED** [review]
Nice work! Just notice some missing docstrings and a minor error a warning messages. Please address them and feel free to merge it then!

**Pull request merged**